luigi/s3.py

```diff
@@ -449,13 +458,14 @@ def move(self, source_path, destination_path, **kwargs):
         self.copy(source_path, destination_path, **kwargs)
         self.remove(source_path)
 
-    def listdir(self, path, start_time=None, end_time=None):
+    def listdir(self, path, start_time=None, end_time=None, with_key=False):
         """
         Get an iterable with S3 folder contents.
         Iterable contains paths relative to queried path.
 
         :param start_time: Optional argument to copy files with modified dates after start_time
         :param end_time: Optional argument to copy files with modified dates before end_time
+        :param with_key: Optional argument, when set to True will return a tuple (filename, boto_key_object)
```

Comment:
Sorry to nitpick, but can you change this to something along the lines of

> Optional argument, when set to True causes function to yield (filename, boto.s3.key.Key) tuples

So we're a bit more explicit in the documentation - thanks!